pkg/model/process_energy.go

```diff
@@ -304,3 +283,29 @@ func addEstimatedEnergy(processIDList []uint64, processesMetrics map[uint64]*sta
 		}
 	}
 }
+
+// addGPUEnergy estimates the GPU power consumption
+func addGPUEnergy(processIDList []uint64, processesMetrics map[uint64]*stats.ProcessStats, nodeMetrics *stats.NodeStats) {
+	// estimate the associated power comsumption of GPU for each process
+	if !gpu.IsGPUCollectionSupported() {
+		return
+	}
+	for gpuID := range gpu.GetGpus() {
+		gpuName := fmt.Sprintf("%s%v", utils.GenericGPUID, gpuID)
+		numberOfProcesses := len(processIDList)
+		// calculate the gpu's processes energy consumption for each gpu
+		processIdlePower := nodeMetrics.EnergyUsage[config.IdleEnergyInGPU].GetDeltaValuesByKey(gpuID) / uint64(numberOfProcesses)
+		dynPower := nodeMetrics.EnergyUsage[config.DynEnergyInGPU].GetDeltaValuesByKey(gpuID)
+		klog.V(6).Infof("GPU %s: idle power: %v, dynamic power: %v\n", gpuName, processIdlePower, dynPower)
+		for pid, v := range processesMetrics {
+			v.EnergyUsage[config.IdleEnergyInGPU].SetDeltaStat(gpuName, processIdlePower)
+			usage := v.ResourceUsage[config.GPUComputeUtilization].GetDeltaValuesByKey(gpuName)
```

Comment:
The usage is not really the usage here right?
This is the GPU ratio....So, we might need to change the metric name `GPUComputeUtilization` to `GPUComputeRatio`.